### ISSUE TITLE
.wav support

### ISSUE THREAD

**Issue by KeitIG** [leetreveil/musicmetadata#104]
Hi,

it seems that .wav do have some metadatas (e.g, taglib parse them https://www.npmjs.com/package/taglib#formats-formats), but musicmetadata throw an error `Uncaught Error: Could not find metadata header`

I know musicmetadata is full JS and taglib use some C++, and this could be why it supports them, but Is there a plan to support .wav ? :)

**Comment by leetreveil** [comment]
There's absolutely no reason why musicmetadata can't handle wav/riff file formats. I'll be happy to help you out with the implementation if you feel like taking it on :)

**Comment by KeitIG** [comment]
Well, after some investigations, it seems like musicmetadata return a correct object, but with empty fields, cause .wav files often don't have metas.

I thought it would be a musicmetadata problem, but it was my bad in my application: I was throwing an error, so the script was stopping. I just used a console.warn instead, and replace the `metadata.title` value by the file name, and the others keys by `'unknown'`

``` javascript
 var file = 'somewhere/something.wav';

mmd(fs.createReadStream(file), { duration: true }, function (err, metadata) {

    if (err) console.warn('An error occured while reading ' + file + ' id3 tags.'); // warn instead of throw err;

    if(metadata.artist.length === 0) metadata.artist = ['Unknown artist'];
    if(metadata.album === null || metadata.album === '') metadata.album = 'Unknown';
    if(metadata.title === null || metadata.title === '') metadata.title = path.parse(file).base;

    // ...
});
```

It's a workaround, but it's doing the job. I'm looking for a .wav with metas to try.

**Comment by KeitIG** [comment]
Yes that was my bad indeed, .wav does not have metas, so I just have to "fake" ones to store them in the database, sorry

**Comment by KeitIG** [comment]
Ok, I'm reopening it since I need to have at least the duration of the .wav file. I'll try to submit a PR soon, must read the code before.

**Comment by KeitIG** [comment]
Hi @leetreveil !

I think I found a way to get the .wav duration. I know that musicmetadata is a lib for audio files with metas, so I don't know if a .wav integration makes a lot of sense. If you think it does, I'll submit a PR. Otherwise we can close this.

**Comment by leetreveil** [comment]
Yeah, I agree it would only be useful if we could parse metadata + duration. Closing! Can you share the code? I'm curious :)

**Comment by KeitIG** [comment]
You can find it here: https://github.com/rackfx/Node-WAV-File-Info/blob/master/wav-file-info.js

Basically, only this line is important: 
```
{
    duration: ((read_result.chunk_size) / (read_result.sample_rate * read_result.num_channels * (read_result.bits_per_sample / 8)))
}
```

just simple operations on the buffer informations.

**Comment by hughrawlinson** [comment]
Wav files can contain metadata in their INFO chunk, it can be have arbitrary encoding, including WMP or ID3 tags. Relevant line of source code in Audacity [here](https://github.com/audacity/audacity/blob/master/src/export/ExportPCM.cpp#L333). Knowing this, would it be possible to reopen this issue and I'll look into figuring out how to provide an implementation? (I'll also try and do an AIFF parser if that's of interest).

**Comment by leetreveil** [comment]
Sure! Reopened.
